packages/react/src/components/connect/ConnectModal.tsx

```diff
@@ -174,16 +174,17 @@ export const ConnectModal = ({
     } else {
       // Handle connecting with social / passkey
       try {
-        wallet = await connect(() =>
-          connectWallet({
-            client,
-            chainId: chain.id,
-            method,
-            authMode,
-            redirectUrl,
-            redirectExternally,
-          }),
-        );
+        const inAppWallet = await connectWallet({
+          client,
+          chainId: chain.id,
+          method,
+          authMode,
+          redirectUrl,
+          redirectExternally,
+        });
+        if (!redirectExternally) {
```

Comment:
Is no `else` handled here because the window has already redirected away? Some comments may help add context around what's happening here